### PR TITLE
chore(circle): bump go to 1.12

### PR DESCRIPTION
**Pull request by bacongobbler**
Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ jobs:
     working_directory: /go/src/k8s.io/helm
     parallelism: 3
     docker:
-      - image: circleci/golang:1.11.5
+      - image: circleci/golang:1.12.0
```
Comment (adamreese) [inline .circleci/config.yml]: maybe use `circleci/golang:1.12`?

**Review by adamreese — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (bacongobbler) [reply .circleci/config.yml]: I was thinking about it. Didn't know if the explicit patch release number was intentional or not so I opted for the least surprise. :)

I'll go change it as suggested.

**Comment by bacongobbler** [comment]
closes #3290

**Comment by adamreese** [comment]
needs a rebase on `dev-v3`

**Pull request merged**